### PR TITLE
ServerAddress.FromUrl() should throw for invalid url (#875)

### PR DESCRIPTION
**Pull request by mikeharder**
When viewing this diff, append `w=1` to the query string to ignore whitespace changes.  Many lines had one level of indent removed.

@halter73, @cesarbs

### CODE REVIEW

**Comment by khellang** [comment]
:shipit:

File: test/Microsoft.AspNetCore.Server.KestrelTests/ServerAddressTests.cs [hunk 1]
```diff
@@ -14,9 +14,9 @@ public class ServerAddressTests
         [InlineData("")]
         [InlineData("5000")]
         [InlineData("//noscheme")]
-        public void FromUriReturnsNullForSchemelessUrls(string url)
+        public void FromUriThrowsForSchemelessUrls(string url)
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.KestrelTests/ServerAddressTests.cs]: We decided to stop allowing "http://", right?

Comment (halter73) [inline test/Microsoft.AspNetCore.Server.KestrelTests/ServerAddressTests.cs]: Nvm. I see the other PR now :smile:

**Comment by halter73** [comment]
:shipit: 

Other reviewers should use the following link for a shorter diff: https://github.com/aspnet/KestrelHttpServer/pull/883/files?w=1

**Comment by cesarbs** [comment]
:ship:

**Pull request merged**